### PR TITLE
Fixed possible race condition

### PR DESCRIPTION
**Pull request by MarconZet**

### CODE REVIEW

File: codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala [hunk 1]
```diff
@@ -84,8 +95,11 @@ class SerializerCheckCompilerPluginComponent(
                 None
               }
             }
-
-            val filterRegex = extractValueOfLiteralConstantFromTree[String](regexTree)
+            val filterRegex =
+              regexTree match {
+                case Select(_, TermName("$lessinit$greater$default$2")) => Some(".*")
```
Comment (MarconZet) [inline codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala:100]: An unrelated bug, where compiler plugin failed to resolve default argument.

File: sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala [hunk 2]
```diff
@@ -19,6 +19,9 @@ object AkkaSerializationHelperPlugin extends AutoPlugin {
 
   override def globalSettings: Seq[Def.Setting[_]] = Nil
 
+  override def buildSettings: Seq[Def.Setting[_]] =
+    Seq(ashCompilerPluginCacheDirectory := baseDirectory.value / "target")
```
Comment (MarconZet) [inline sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala:23]: This resolves to `ThisBuild / ashCompilerPluginCacheDirectory := ThisBuild / baseDirectory.value / "target"`

Comment (PawelLipski) [reply sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala:23]: Okay... so AFAICS now both `codec-registration-checker-cache.csv` and `dump-persistence-schema-cache/` are only located in top-level `target/`, and **not** in `target/` of sbt modules where ash is enabled...

Comment (PawelLipski) [reply sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala:23]: Would it be possible to address https://github.com/VirtusLab/akka-serialization-helper/issues/130 by also using top-level `target/` as the location for `akka-persistence-schema.yaml`? Coz having the same yaml duplicated across N different locations (for each module that enables ash) is a bit awkward&confusing

File: sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala [hunk 3]
```diff
@@ -49,7 +51,7 @@ object AkkaSerializationHelperPlugin extends AutoPlugin {
         new File(
           (ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputDirectoryPath).value) / (ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputFilename).value,
       ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputFilename := s"${name.value}-dump-persistence-schema-${version.value}.yaml",
-      ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputDirectoryPath := target.value.getPath) ++
+      ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputDirectoryPath := ashCompilerPluginCacheDirectory.value.getPath) ++
```
Comment (MarconZet) [inline sbt-akka-serialization-helper/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala:54]: @PawelLipski a change for one dump file

**Review by PawelLipski — APPROVED** [review]
Ok! tested on Laguna repo

**Pull request merged**